src/main/scala/io/picnicml/doddlemodel/metrics/ClassificationMetrics.scala

```diff
@@ -42,4 +42,30 @@ object ClassificationMetrics {
       numTp / (numTp + (!yPredPositive &:& yPositive).activeSize.toDouble)
     }
   }
+
+  /** F1 score. */
+  object F1Score extends Metric {
+
+    override lazy val higherValueIsBetter: Boolean = true
+
+    override def apply(y: Target, yPred: Target): Double = {
+      require(y.length == yPred.length, "Target vectors need to be of equal length")
+      require(numberOfTargetClasses(y) == 2, "F1 score is defined for a binary classification task")
+      val prec = precision(y, yPred)
+      val rec = recall(y, yPred)
+
+      2 * (prec * rec) / (prec + rec)
```

Comment:
Really nitpicking here but I usually like to write constants that are converted to doubles as `2.0` for example, at least my mental model is that as soon as I see the double format I know that the evaluated expression will be of the type `Double`. The same as you did for the Hamming loss.